synapse/storage/database.py

```diff
@@ -1560,6 +1560,21 @@ def simple_select_one_onecol_txn(
     ) -> Optional[Any]:
         ...
 
+    @overload
+    @classmethod
+    def simple_select_one_onecol_txn(
+        cls,
+        txn: LoggingTransaction,
+        table: str,
+        keyvalues: Dict[str, Any],
+        retcol: str,
+        allow_none: bool = False,
+    ) -> Optional[int]:
+        ret = cls.simple_select_one_onecol_txn(cls, txn, table=table, keyvalues=keyvalues, retcol=retcol, allow_none=allow_none)
+        if ret:
+            return int(ret)
+        return ret
```

Comment:
This doesn't look right --- `@overloads` should not have an implementation; they should be for type hints only. See https://mypy.readthedocs.io/en/stable/more_types.html#function-overloading for more details.

Furthermore this is a generic function and there is no particular reason this should return an int. Additionally, the call to `cls.simple_select_one_onecol_txn` would infinitely recurse.

Can you explain your thinking here? What error was this attempting to address?